### DISCUSSION TITLE
Joyride overlay flickering, flashing between steps

### DISCUSSION THREAD

**Discussion by magda5281** [gilbarbara/react-joyride#872] [Q&A] [answered]
Hi, 
I am new to Joyride. 
I have seen that someone asked about that issue -Joyride overlay disappears between steps causing the flashing/ jumpy effect - some time ago but I couldn't find a solution anywhere. Could anyone advise if that was solved anywhere?

Thank you.
Magda

https://user-images.githubusercontent.com/80685460/208112413-8508a0e9-b00a-4658-a65e-aaa0bab4aa8f.mov

**Comment by gilbarbara** [comment, marked as answer]
Hey @magda5281 

Is this happening on production too or just development? Are you using StrictMode?
Does the flickering happen on https://react-joyride.com/?

**Comment by magda5281** [comment]
Hi,

No on  https://react-joyride.com/ it is not happening. Which is why I
thought I might be doing something wrong in my local project.

Magda

**Comment by evansking** [comment]
Been 4 month, but did you get to the bottom of this? Same issue here.

**Comment by evansking** [comment]
Looks like StrictMode is the issue

**Reply by juniorforlife** [reply]
@evansking @magda5281 I’m facing the issue. Apart from StrictMode, setTimeout to move to next steps will also cause the flickering

**Comment by blacktail** [comment]
It's true, StrictMode cause this issue. React will force to do twice rendering in StrictMode

**Comment by kaleb110** [comment]
turn off strict mode in next.config.js in development.

**Reply by Amoji09** [reply]
This worked for me. Thank you!

**Comment by maoryadin** [comment]
@gilbarbara 
@kaleb110 
I can confirm that the problem is occurs when there is a setTimeout, using `react-joyride@latest` is not resolve the issue:
example:
```
      // Add input event listener for input/select elements to trigger validation dynamically.
      if (targetElement instanceof HTMLInputElement) {
        const handleInput = () => {
          if (step.data.validateStep?.(targetElement)) {
            targetElement.removeEventListener('input', handleInput)
            setTimeout(() => {
              activeTour.tourActions.next()
            }, 0)
          } else {
            setShowHint(true)
          }
        }
        targetElement.addEventListener('input', handleInput)
        return () => targetElement.removeEventListener('input', handleInput)
      }
    } else {
      // If no validation is required, attach a click handler to proceed.
      const handleClick = () => {
        setTimeout(() => {
          activeTour.tourActions.next()
        }, 0)
      }
      targetElement.addEventListener('click', handleClick)
      return () => targetElement.removeEventListener('click', handleClick)
    }
  }, [targetElement, step.data.validateStep, activeTour.tourActions])

```

**Comment by novirusallowed** [comment]
Same issue. Happens in production too with strictMode off

**Comment by pablojsx** [comment]
Same issue, using nextjs 14
